src/sentry/api/endpoints/organization_events_stats.py

```diff
@@ -47,13 +105,18 @@ def get(self, request, organization):
             limit=10000,
         )
         serializer = SnubaTSResultSerializer(organization, None, request.user)
-        return Response(
-            serializer.serialize(
-                snuba.SnubaTSResult(result, snuba_args["start"], snuba_args["end"], rollup)
-            ),
-            status=200,
+
+        snuba_data = serializer.serialize(
+            snuba.SnubaTSResult(result, snuba_args["start"], snuba_args["end"], rollup)
         )
 
+        if should_cache:
+            cache_timeout = 900  # 15 minutes = 900 seconds
+            cache.set(cache_key, snuba_data, cache_timeout)
+            logging.info("foo cache miss: %s", cache_key)
+
+        return Response(snuba_data, status=200)
```

Comment:
Thoughts on having this return `snuba_data` directly instead of wrapping it in a response object? Makes `get_stats` more testable, and would mean that you can append data directly [here](https://github.com/getsentry/sentry/pull/15659/files#diff-da22baf05fb18f930a4f20f387348057R56)